draft-barnes-acme.md

```diff
@@ -1861,12 +1859,12 @@ The client serializes the validation object to UTF-8, then uses its account
 private key to sign a JWS with the serialized JSON object as its payload.  This
 JWS is NOT REQUIRED to have the "nonce" header parameter.
 
-The record provisioned to the DNS is the "signature" value from the JWS, i.e.,
-the base64-encoded signature value.  The client constructs the validation domain
-name by appending the label "_acme-challenge" to the domain name being
-validated, then provisions a TXT record with the signature value under that
-name. For example, if the domain name being validated is "example.com", then the
-client would provision the following DNS record:
+The record provisioned to the DNS is the base64-encoded SHA-256 of the JWS
```

Comment:
The challenge payload doesn't contain the public portion of the account key which I believe is necessary.  I was also trying to maintain some uniformity with the existing SimpleHTTP challenge (provide the full validation JWS). 

Here is one attack that I was trying to protect against.

Imagine a client communicating with an "evil" ACME server (or any untrustworthy TLS endpoint).  The "evil" ACME server could attain authorization over a domain it doesn't own with its own public key from a legitimate ACME server.

The evil ACME server would forward requests to the true ACME server except swap out its own public key for all of the requests.  When the authorization step came, the client would unwittingly allow the evil ACME server to use the same validation challenge.